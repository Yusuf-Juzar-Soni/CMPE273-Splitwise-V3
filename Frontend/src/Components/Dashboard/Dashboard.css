
.button-settleup {
  
  background: orange !important ;
  padding: 10px;
  text-align: center;
  border-radius: 5px;
  color:black;
  font-weight: bold;
  line-height: 25px;
}
