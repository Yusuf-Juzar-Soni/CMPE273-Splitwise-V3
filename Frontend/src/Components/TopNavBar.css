.dtoggle{
    margin-right: 109px;
}
