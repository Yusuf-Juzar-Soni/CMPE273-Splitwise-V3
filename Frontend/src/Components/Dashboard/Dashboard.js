import React, { useEffect, useState } from "react";
import { useSelector } from "react-redux";
import "./Dashboard.css";
import LeftNavBar from "../LeftNavBar/LeftNavBar";
import TopNavBar from "../TopNavBar";
import DashboardInfo from "../DashboardInfo";

const Dashboard = function () {
  
  const isLogged = useSelector((state) => state.isLogged.username);
  console.log("hello");
  return (
    <div className="dashboard">
      <div>
        <TopNavBar />
      </div>
      <h4 data-testid="Dashboard">Dashboard</h4>

      <div className="row">
        <div div className="col-md-2">
          <LeftNavBar />
        </div>
        <div div className="col-md-10">
          <DashboardInfo />
        </div>
      </div>
    </div>
  );
};

export default Dashboard;
