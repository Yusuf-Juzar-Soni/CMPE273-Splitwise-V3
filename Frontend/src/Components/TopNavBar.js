import React from "react";
import Navbar from "react-bootstrap/Navbar";
import Nav from "react-bootstrap/Nav";
import Dropdown from "react-bootstrap/Dropdown";
import { useHistory } from "react-router-dom";
import { ListGroup, Button } from "react-bootstrap";
import { useSelector } from "react-redux";
const queryString = require("query-string");
import { useLocation } from "react-router-dom";
import bg_image0 from "./assets/login_logo.png";
import "./TopNavBar.css";

function TopNavBar() {
  const history = useHistory();
  const location = useLocation();
  const parsed = queryString.parse(location.search);
  const isLogged = useSelector((state) => state.isLogged.username);

  function redirectToProfile(emailId) {
    history.push({
      pathname: "/profile",
      search: `?email=` + emailId,
    });
  }

  function redirectToLogin() {
    localStorage.clear();
    history.push({
      pathname: "/login",
    });
  }
  return (
    <div>
      <Navbar bg="success" expand="lg">
        <Navbar.Brand href="#home">
          <img src={bg_image0} width="50" height="50"></img>
        </Navbar.Brand>
        <Nav className="mr-auto">
          <Nav.Link href="#home" />
        </Nav>
        <Dropdown>
          <Dropdown.Toggle
            className="dtoggle"
            variant="info"
            id="dropdown-basic"
          >
            {isLogged}
          </Dropdown.Toggle>
          <Dropdown.Menu>
            <Dropdown.Item
              className="ditem"
              onClick={(event) => redirectToProfile(parsed.email)}
            >
              Profile
            </Dropdown.Item>
            <Dropdown.Item onClick={(event) => redirectToLogin()}>
              Logout
            </Dropdown.Item>
          </Dropdown.Menu>
        </Dropdown>
      </Navbar>
    </div>
  );
}

export default TopNavBar;
