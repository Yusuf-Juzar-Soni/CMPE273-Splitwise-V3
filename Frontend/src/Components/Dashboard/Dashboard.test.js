import React from "react";
import { render, cleanup } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import { createStore } from "redux";
import { MemoryRouter } from "react-router-dom";
// import configureStore from 'redux-mock-store';
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import Dashboard from "./Dashboard";
import combReducers from "../../reducers";
import TopNavBar from "../TopNavBar";

const store = createStore(combReducers);

const component = TestRenderer.create(
  <Provider store={store}>
    <MemoryRouter>
      <Dashboard>
        <TopNavBar />
      </Dashboard>
    </MemoryRouter>
  </Provider>
);

afterEach(cleanup);

it("renders", async () => {
  expect(component.toJSON()).toMatchSnapshot();
});

test("Check for header", async () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Dashboard>
          <TopNavBar />
        </Dashboard>
      </MemoryRouter>
    </Provider>
  );
  expect(getByTestId("Settle Up")).toHaveTextContent("Settle Up");
});
