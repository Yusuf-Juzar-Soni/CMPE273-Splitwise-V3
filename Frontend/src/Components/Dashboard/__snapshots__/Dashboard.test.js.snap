// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="dashboard"
>
  <div>
    <div>
      <nav
        className="navbar navbar-expand-lg navbar-light bg-success"
      >
        <a
          className="navbar-brand"
          href="#home"
        >
          <img
            height="50"
            src="login_logo.png"
            width="50"
          />
        </a>
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-link"
            data-rb-event-key="#home"
            href="#home"
            onClick={[Function]}
            onKeyDown={[Function]}
          />
        </div>
        <div
          className="dropdown"
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dtoggle dropdown-toggle btn btn-info"
            disabled={false}
            id="dropdown-basic"
            onClick={[Function]}
            type="button"
          >
            
          </button>
        </div>
      </nav>
    </div>
  </div>
  <h4
    data-testid="Dashboard"
  >
    Dashboard
  </h4>
  <div
    className="row"
  >
    <div
      className="col-md-2"
      div={true}
    >
      <div>
        <div>
          <div
            className="flex-column nav"
            color="red"
            onKeyDown={[Function]}
          >
            <a
              className="nav-link"
              data-rb-event-key={null}
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              Dashboard
            </a>
            <a
              className="nav-link"
              data-rb-event-key={null}
              data-testid="Activity"
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              Activity
            </a>
            <a
              className="nav-link"
              data-rb-event-key={null}
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              Groups
            </a>
            <input
              onChange={[Function]}
              type="text"
            />
            <div
              className="list-group-design list-group"
              onKeyDown={[Function]}
            />
            <a
              className="nav-link"
              data-rb-event-key={null}
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              Create Group
            </a>
            <a
              className="nav-link"
              data-rb-event-key={null}
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              Invite List
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-md-10"
      div={true}
    >
      <div>
        <div
          className="row"
        >
          <div
            className="row"
          >
            <div
              className="col-md-12"
            >
              <button
                className="button-settleup btn btn-primary btn-sm"
                data-testid="Settle Up"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Settle Up
              </button>
              <button
                className="button-creategroup btn btn-primary btn-sm"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Create Group
              </button>
            </div>
          </div>
        </div>
        <br />
        <br />
        <div
          className="row"
        />
        <div
          class="row justify-content-center"
        >
          <title>
            Welcome to Dashboard
          </title>
          <h4>
            Welcome to Dashboard
          </h4>
        </div>
        <div
          className="row mt-2"
        >
          <div
            className="col-md-4"
          >
            <b>
              YOU OWE: $
              0
            </b>
          </div>
          <div
            className="col-md-4"
          >
            <b>
              YOU ARE OWED: $
              0
            </b>
          </div>
          <div
            className="col-md-4"
          >
            <b>
              BALANCE: $
              0
            </b>
          </div>
        </div>
        <div
          className="show-grid"
        >
          <div
            className="row mt-4"
          >
            <div
              className="col-md-6"
            >
              <p>
                <center>
                  <b>
                    <u>
                      YOU OWE
                    </u>
                  </b>
                </center>
              </p>
              <div
                className="list-group"
                onKeyDown={[Function]}
              />
            </div>
            <div
              className="col-md-6"
            >
              <p>
                <center>
                  <b>
                    <u>
                      YOU ARE OWED
                    </u>
                  </b>
                </center>
              </p>
              <div
                className="list-group"
                onKeyDown={[Function]}
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
